File: backend/src/main/java/com/ar/backend/dtos/TransactionDto.java
package com.ar.backend.dtos;

import com.ar.backend.entities.Account;
import com.ar.backend.enums.TransactionStatus;
import com.ar.backend.enums.TransactionType;
import com.fasterxml.jackson.annotation.JsonFormat;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class TransactionDto {

  private Long id;
  private Account.AccountInfoDto account;
  private Long transactionCode;
  private TransactionType type;
  @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
  private LocalDateTime date;
  private BigDecimal amount;
  private TransactionStatus transactionStatus;
}

File: frontend/src/app/core/models/transaction.dto.ts
import { AccountInfoDto } from './account.dto';

export interface TransactionDto {
  id: number;
  account: AccountInfoDto;
  transactionCode: number;
  type: TransactionType;
  date: Date;
  amount: number;
  transactionStatus: TransactionStatus;
}

export enum TransactionStatus {
  ACCEPTED = 'ACCEPTED',

  REJECTED = 'REJECTED',
}

export enum TransactionType {
  DEPOSIT = 'DEPOSIT',

  TRANSFER = 'TRANSFER',

  WITHDRAW = 'WITHDRAW',
}
